PennMobile/src/main/java/com/pennapps/labs/pennmobile/MainActivity.java

```diff
@@ -309,4 +308,71 @@ public void removeTabs() {
             appBar.removeViewAt(1);
         }
     }
+
+    private void getPermission(int code) {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) !=
+                PackageManager.PERMISSION_GRANTED ||
+                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) !=
+                        PackageManager.PERMISSION_GRANTED) {
+
+            ActivityCompat
+                    .requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION,
+                            Manifest.permission.ACCESS_COARSE_LOCATION}, code);
+        } else {
+            Fragment fragment = null;
+            switch(code) {
+                case CODE_MAP:
+                    fragment = new MapFragment();
+                    break;
+                case CODE_TRANSIT:
+                    fragment = new TransitFragment();
+                    break;
+            }
+            fragmentTransact(fragment);
+        }
+    }
+
+    private void fragmentTransact(Fragment fragment) {
+        if (fragment != null) {
+            FragmentManager fragmentManager = getSupportFragmentManager();
+            fragmentManager.beginTransaction()
+                    .replace(R.id.content_frame, fragment)
+                    .addToBackStack(null)
+                    .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+                    .commit();
+        }
+        mDrawerLayout.closeDrawer(mDrawerList);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        if (grantResults.length == 0 || grantResults[0] == PackageManager.PERMISSION_DENIED) {
+            Toast.makeText(this, "Access of your location is required to use this feature.", Toast.LENGTH_LONG).show();
```

Comment:
Suggestion for toast: "Location permissions are required to use the Penn Map, please try again."